qualang_tools/control_panel/video_mode/README.md

```diff
@@ -0,0 +1,146 @@
+    x_element="output_ch1",  # Must be a valid QUA element
+    y_element="output_ch2",  # Must be a valid QUA element
+    integration_time=10e-6,  # Integration time in seconds
+    readout_element="measure_ch",  # Must be a valid QUA element
+    readout_pulse="readout",  # Name of the readout pulse registered in the readout_element
+)
+```
+Note that this `InnerLoopAction` assumes that the `readout_pulse` has two integration weights called `cos` and `sin`
+
+Next we define the sweep axes, which define the values that the 2D scan will take as coordinates.
+
+```python
+from qualang_tools.control_panel.video_mode.sweep_axis import SweepAxis
+x_axis = SweepAxis(
+    name="voltage_gate1",  # Sweep axis name, used among others for plotting
+    span=0.03,  # Span of the sweep in volts
+    points=51,  # Number of points to sweep
+)
+y_axis = SweepAxis(name="voltage_gate2", span=0.03, points=51)
+```
+The `SweepAxis` contains additional attributes, such as attenuation and a voltage offset, the latter of which is described in `Advanced Usage`.
+
+Next we define the data acquirer, which is the object that will handle the data acquisition.
+```python
+from qualang_tools.control_panel.video_mode.data_acquirer import OPXDataAcquirer
+data_acquirer = OPXDataAcquirer(
+    qm=qm,
+    qua_inner_loop_action=inner_loop_action,
+    scan_mode=scan_mode,
+    x_axis=x_axis,
+    y_axis=y_axis,
+)
+```
+
+You can now test the data acquirer before using it in video mode.
+```python
+data_acquirer.run_program()
+results = data_acquirer.acquire_data()
+```
+
+Finally, we can start the video mode.
+```python
+from qualang_tools.control_panel.video_mode.video_mode import VideoMode
+video_mode = VideoMode(data_acquirer=data_acquirer)
+video_mode.run()
+```
+
+Note that if you want to run this code in an interactive environment such as a Jupyter notebook, you should use `video_mode.run(use_reloader=False)`.
+
+You can now access video mode from your browser at `http://localhost:8050/` (the port may be different, see the output logs for details).
+
+
+## Advanced Usage
+### Voltage offsets
+
+The `SweepAxis` class has an `offset_parameter` attribute, which is an optional parameter that defines the sweep offset. This can be a QCoDeS DC voltage source parameter or a `VoltageParameter` object.
+
+As an example, let us assume that we have a QCoDeS parameter `x_gate` for the DC voltage of a gate:
+
+```python
+x_offset()  # Returns the DC voltage, e.g. 0.62
+```
+
+In this case, we can pass this parameter to the `SweepAxis` class to define the sweep offset.
+```python
+x_axis = SweepAxis(name="gate", span=0.03, points=51, offset_parameter=x_offset)
+```
+The video mode plot should now correctly show the sweep axes with the correct offset.
+
+Note that if the offset voltage is changed, it will need to be changed in the same kernel where the video mode is running. One solution for this is using the `VoltageControl` module in py-qua-tools.
```

Comment:
This is from another PR right?
Is it also ready for review?

Reply:
I still don't understand how this works.
When VideoMode is running the kernel is being used so that you can't run anything there...
One solution could be to use what we did here: https://github.com/qua-platform/py-qua-tools/tree/main/qualang_tools/video_mode

Reply:
This doesn't work, it needs external instrument calls. The way it works is as follows:

Two threads are created:
**Video Mode dash app** that runs an OPX program and repeatedly shows results
**Voltage Control GUI** that allows setting QDAC voltages